src/transactions.ts

```diff
@@ -45,6 +45,14 @@ const stateMachine: { [state in TxnState]: TxnState[] } = {
   ]
 };
 
+const ACTIVE_STATES: TxnState[] = [TxnState.STARTING_TRANSACTION, TxnState.TRANSACTION_IN_PROGRESS];
```

Comment:
Should these be Sets and we can use has? the teensiest speed improvement.

Reply:
Changed to sets.